created Application Form Component

DeniedPath opened the pull request:
Users are now able to apply by clicking the apply button

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a seamless job application experience with a modal window.
  - Updated “Apply Now” buttons now open the modal for submitting applications.
  - Added an interactive job application form allowing users to input details, upload documents, and receive success confirmation upon submission.
  - Added a "Recent Applicants" section in the admin dashboard to display applicant information.
  - Streamlined job listing functionality for a more user-friendly interface.
  - Enhanced drag-and-drop interactions in the applicant dashboard with visual feedback.
  - Improved job filters with multi-select dropdowns for better user interaction.
  - Added a 404 error page to enhance user navigation experience.
  - Implemented stricter ESLint and TypeScript checks during builds for improved code quality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

DeniedPath committed:
created Application Form Component
Users are now able to apply by clicking the apply button

AD-Archer commented on app/applicant/jobs/page.tsx:
```diff
@@ -155,6 +158,20 @@ export default function ApplicantJobListings() {
+    // Here you would typically send the data to your backend
+    setApplicationSubmitted(true);
+    // Don't close the modal yet, show success message first
+  };
```
@DeniedPath why is this console log in pull?

AD-Archer commented on components/job-application-form.tsx:
```diff
@@ -0,0 +1,222 @@
+
+interface JobApplicationFormProps {
+  job: any
+  onSubmit: (formData: any) => void
```
you also have an any type which means this code won't complie correctly and if it does that means that we may run into a type error @DeniedPath

AD-Archer committed:
Updates to CI/CD workflow

-remove next.config.ts as it is unused and not proper

-create ci.yml to ensure that pull requests must past a lint, build and typecheck

-update package.json to ensure that typecheck is a script

AD-Archer committed:
docs: add Code of Conduct for Pull Requests and Code Submissions

AD-Archer committed:
fix a ton of lint errors

AD-Archer committed:
fix 10 lint errors

AD-Archer committed:
fix all lint errors

AD-Archer committed:
fix applicat jobs

AD-Archer committed:
fix applicant dashboard

AD-Archer committed:
fix admin dashboard

AD-Archer committed:
minor changes(i need to rebase)

AD-Archer committed:
remove toasts fix dashboards

AD-Archer committed:
refactor: update applicant job application form and remove unused code; add NotFound component

AD-Archer committed:
update images on the platform

AD-Archer committed:
update SEO

AD-Archer committed:
update footer

AD-Archer committed:
Merge branch 'main' into Kristian

Signed-off-by: Archer <adarcher21@gmail.com>

AD-Archer committed:
fix toast
